CommonTools/PileupAlgos/plugins/PuppiProducer.cc

```diff
@@ -103,7 +123,33 @@ void PuppiProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
     int    pVtxId = -9999; 
     bool lFirst = true;
     const pat::PackedCandidate *lPack = dynamic_cast<const pat::PackedCandidate*>(&(*itPF));
-    if(lPack == 0 ) {
+    if (fUsePVAssignmentMap) {
+      const reco::PFCandidate *pPF = dynamic_cast<const reco::PFCandidate*>(&(*itPF));
```

Comment:
Why?  It's assumed that if fUsePVAssignment map is configured to be true, then the input collection is also consistently a collection of PFCandidates and not packed candidates.